python/pecos/qeclib/steane/steane_class.py

```diff
@@ -97,117 +97,61 @@ def __init__(self, name: str, default_rus_limit: int = 3, ancillas: QReg | None
 
         self.default_rus_limit = default_rus_limit
 
-    def p(self, state: str, rus_limit: int | None = None):
+    def p(self, state: str, reject: Bit | None = None, rus_limit: int | None = None):
         """Prepare a logical qubit in a logical Pauli basis state."""
-        if rus_limit is None:
-            rus_limit = self.default_rus_limit
-
         match state:
-            case "+X" | "X":
-                return self.px(rus_limit=rus_limit)
-            case "-X":
-                return self.pnx(rus_limit=rus_limit)
-            case "+Y" | "Y":
-                return self.py(rus_limit=rus_limit)
-            case "-Y":
-                return self.pny(rus_limit=rus_limit)
-            case "+Z" | "Z":
-                return self.pz(rus_limit=rus_limit)
-            case "-Z":
-                return self.pnz(rus_limit=rus_limit)
+            case "|0>" | "+Z" | "Z":
+                state = "+Z"
+            case "|1>" | "-Z":
+                state = "-Z"
+            case "|+>" | "+X" | "X":
+                state = "+X"
+            case "|->" | "-X":
+                state = "-X"
+            case "|+i>" | "+Y" | "Y":
+                state = "+Y"
+            case "|-i>" | "-Y":
+                state = "-Y"
             case _:
                 msg = f"State {state} is not implemented!"
                 raise NotImplementedError(msg)
 
-    def px(self, rus_limit: int | None = None):
-        """Prepare logical |+X>, a.k.a. |+>"""
-
-        if rus_limit is None:
-            rus_limit = self.default_rus_limit
-
-        return PrepRUS(
+        block = PrepRUS(
             q=self.d,
             a=self.a[0],
             init=self.verify_prep[0],
-            limit=rus_limit,
-            state="+X",
+            limit=rus_limit or self.default_rus_limit,
+            state=state,
+            reject=reject or self.scratch[2],
```

Comment:
`reject` kwarg is no longer supported in this version. Looks like this is leading to an error in the tests.

```suggestion
```

Reply:
Oops.  Fixed 🙂